cluster-autoscaler/cloudprovider/azure/azure_manager.go

```diff
@@ -130,14 +130,11 @@ func createAzureManagerInternal(configReader io.Reader, discoveryOpts cloudprovi
 		Cap:      10 * time.Minute,
 	}
 
-	if err := manager.forceRefresh(); err != nil {
-		err = kretry.OnError(retryBackoff, retry.IsErrorRetriable, func() (err error) {
-			return manager.forceRefresh()
-		})
-		if err != nil {
-			return nil, err
-		}
-		return manager, nil
+	err = kretry.OnError(retryBackoff, retry.IsErrorRetriable, func() (err error) {
+		return manager.forceRefresh()
+	})
+	if err != nil {
+		return nil, err
 	}
```

Comment:
https://github.com/comtalyst/autoscaler/commit/0d5a71e867c0c3f9b8207b582b16cbcd0745bf08

But looks ok, could pull the improved version in.